src/library/scala/concurrent/Future.scala

```diff
@@ -731,16 +732,24 @@ object Future {
     val i = futures.iterator
     if (!i.hasNext) Future.never
     else {
+      val deregs = ListBuffer.empty[() => Unit]
+      val l = new CountDownLatch(1)
```

Comment:
Scala.js doesn't have `CountDownLatch`, because it has an inherently blocking API. So as is, this change will break Scala.js. The alternative proposed by @axel22, with an atomic reference, would be fine on Scala.js.